overlays/resctl-demo/home/demo/start-resctl-bench

```diff
@@ -78,18 +78,28 @@ sudo touch ${LOCKFILE}
   sudo sh -c "echo ${BASHPID} > ${LOCKFILE}"
 
 # Create benchmark using resctl-bench & convert results to PDF/TXT
+  SUCCESS=false
   for ((i = 0; i < 10; i++)); do
     if sudo --preserve-env resctl-bench -r "$RESULT_JSON" --logfile=$LOG_FILE run iocost-tune; then
+      SUCCESS=true
       break
     fi
   done
 
   sudo rm -rf $LOCKFILE
-  sudo resctl-bench -r "$RESULT_JSON" format iocost-tune:pdf=$RESULT_PDF >/dev/null
-  sudo resctl-bench -r "$RESULT_JSON" format | sudo tee "$RESULT_SUMMARY" >/dev/null
-
-  echo "resctl-bench finshed successfully!"
-  echo "Results saved to USB stick as $MODEL/$FILENAME"
+  if [[ "${SUCCESS}" == "true" ]]; then
+    sudo resctl-bench -r "$RESULT_JSON" format iocost-tune:pdf=$RESULT_PDF >/dev/null
+    sudo resctl-bench -r "$RESULT_JSON" format | sudo tee "$RESULT_SUMMARY" >/dev/null
+
+    echo "resctl-bench finshed successfully!"
+    echo "Results saved to USB stick as $MODEL/$FILENAME"
+  else
+    sudo resctl-bench -r "${RESULT_JSON}" pack
```

Comment:
The tarball is copied in next line (see below `sudo cp "${RESULT_JSON}" $LOG_DIR/`).  
Actually tarball is created in "${RESULT_JSON}" directory which is inside flasher USB disk in both success and failure cases automatically. 

Thus in resctl-bench success cases , it is part of result/output and in failed cases it is part of LOG_DIR.

Let me know if meant something else. Thanks

Reply:
> This creates the tarball in the current directory. If not copied to the usb stick, this doesn't do anything.

> This creates the tarball in the current directory. If not copied to the usb stick, this doesn't do anything.

The tarball is copied in next line (see below `sudo cp "${RESULT_JSON}" $LOG_DIR/`).  
Actually tarball is created in "${RESULT_JSON}" directory which is inside flasher USB disk in both success and failure cases automatically. 

Thus in resctl-bench success cases , it is part of result/output and in failed cases it is part of LOG_DIR.

Let me know if meant something else. Thanks

Reply:
I see. Isn't that still a bit weird tho? If we want the result file to be in the log directory after a failure, shouldn't the tarball which is there to assist debugging also be in the log directory rather than the result directory?

Reply:
`resctl-bench pack` generate single tarball  which contain both  result + report files on failure, And on success it has only result .
So overall there is no separate debugging tarball  generated. It is same as result tarball in failure.